### ISSUE TITLE
Consider minifying json shipped in package

### ISSUE THREAD

**Issue by SimonSchick** [googleapis/google-api-python-client#1967]
tl;dr we are deploying the SDK to aws lambdas and ran into a scenario where our primary layer became too large, initially we just stripped individual json files from discovery_cache/documents since this is by far the largest culprit (according to `ncdu`).

I also noticed that all of these definitions are actually pretty printed json.

I'm suggesting 2 things:
1. Ship a utility to select (and delete other) json definitions that the user wants to use in a deployment.
2. Minify all json contents (see trivial script below), this reduced the package size (on disk) by approximately 20mb (and probably also improves load times although I did not test for that).

```js
fs.readdirSync('.').forEach(f => {
  const data = JSON.parse(fs.readFileSync(f, 'utf8'));
  fs.writeFileSync(f, JSON.stringify(data));
  console.log('processed', f);
});
```

#### Environment details

  - OS type and version: irrelevant
  - Python version: 3.9
  - pip version: 22.2.2
  - `google-api-python-client` version: 2.64.0

#### Steps to reproduce

  1. Install package
  2. 72mb via `ncdu`

#### Code example

n/a

#### Stack trace

n/a

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

**Comment by tgbugs** [comment]
A related issue on the aws side: https://github.com/boto/botocore/issues/2365.

**Comment by jorenham** [comment]
At the moment the json files take 81.3MB on disk, which is a significant portion of my docker images.
